File: ios/FaithdroidApp/ViewController.m
//
//  ViewController.m
//  FaithdroidApp
//
//  Created by steven zacker on 6/28/19.
//  Copyright © 2019 steven zacker. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController
@synthesize ui;
@synthesize w;
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the
    w = [[FaithdroidMainWindow alloc] init];
    ui = [[UIObjectCBridge alloc] initWithParm:self];
    [w setUI:ui];
    [w onCreate];
}


@end

File: example-ios-project/iFaithApp/ViewController.swift
//
//  ViewController.swift
//  iFaithApp
//
//  Created by asd on 2018/10/25.
//  Copyright © 2018 asd. All rights reserved.
//

import UIKit
import Faithdroid
class ViewController: UIViewController {
    private var a:FaithdroidMainActivity!
    private var parentBridge:UIBridge!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        a = FaithdroidMainActivity()
        parentBridge = UIBridge().create(viewController: self, rootView: addContainer(), factivity: nil)
        a.setUIInterface(parentBridge)
        a.onCreate()
    }

    func addContainer()->UIView {
        let navHeight = Int(UIApplication.shared.statusBarFrame.height) + Int(navigationController!.navigationBar.frame.height)
        let rootView = UIView()
        rootView.clipsToBounds = true
        let rWidth = Int(self.view.frame.width)
        let rHeight = Int(self.view.frame.height) - navHeight
        rootView.frame = CGRect(x: 0, y: navHeight, width: rWidth, height: rHeight)
        rootView.backgroundColor = UIColor.white
        self.view.addSubview(rootView)
        return rootView
    }
}

